src/js/apps/patients/patient/archive/archive_app.js

```diff
@@ -13,15 +13,63 @@ export default App.extend({
   beforeStart({ patient }) {
     const currentWorkspace = Radio.request('workspace', 'current');
     const states = currentWorkspace.getStates();
-    const filter = { states: states.groupByDone().done.getFilterIds() };
+
+    this.filter = { states: states.groupByDone().done.getFilterIds() };
 
     return [
-      Radio.request('entities', 'fetch:actions:collection:byPatient', { patientId: patient.id, filter }),
-      Radio.request('entities', 'fetch:flows:collection:byPatient', { patientId: patient.id, filter }),
+      Radio.request('entities', 'fetch:actions:collection:byPatient', { patientId: patient.id, filter: this.filter }),
+      Radio.request('entities', 'fetch:flows:collection:byPatient', { patientId: patient.id, filter: this.filter }),
     ];
   },
   onStart({ patient }, actions, flows) {
-    this.collection = new Backbone.Collection([...actions.models, ...flows.models]);
+    // remove these when BE api endpoint `filter[state]=` is fixed
+    const doneActions = actions.filter(action => action.isDone());
+    const doneFlows = flows.filter(flow => flow.isDone());
+
+    this.collection = new Backbone.Collection([...doneActions, ...doneFlows]);
+
+    this.subscribe();
+
     this.showChildView('content', new ListView({ collection: this.collection }));
   },
+  subscribe() {
+    const channel = Radio.channel('ws');
+
+    channel.request('subscribe', this.collection.models, {
+      filters: { actions: this.filter, flows: this.filter },
+    });
+
+    this.listenTo(channel, 'message', ({ category, resource, payload }) => {
+      const modelResource = category === 'ResourceCreated' ? payload.resource : resource;
+      const model = Radio.request('entities', 'get:store', modelResource);
+
+      if (this.collection.get(model) || model.isFetching()) return;
+
+      if (modelResource.type === 'flows') {
```

Comment:
yeah I'm perfectly fine combining these with `fetch:${ model.type }:model`   we can probably recombine these methods now since they're simpler than the worklist

Reply:
in fact..  I think we will _always_ have the context needed that this can be
```js
const model = Radio.request('entities', 'get:store', modelResource);

if (this.collection.get(model) || model.isFetching()) return;
model.fetch().then(() => { 
  this.collection.add(model);

  Radio.request('ws', 'add', model);
});
```

unless I'm missing something?

Reply:
Yup that works 👍